Xamarin.Forms.Platform.iOS/CollectionView/ObservableItemsSource.cs

```diff
@@ -44,18 +48,38 @@ void CollectionChanged(object sender, NotifyCollectionChangedEventArgs args)
 
 		void Move(NotifyCollectionChangedEventArgs args)
 		{
-			var oldPath = NSIndexPath.Create(0, args.OldStartingIndex);
-			var newPath = NSIndexPath.Create(0, args.NewStartingIndex);
+			var count = args.NewItems.Count;
 
-			_collectionView.MoveItem(oldPath, newPath);
+			if (count == 1)
+			{
+				// For a single item, we can use MoveItem and get the animation
+				var oldPath = NSIndexPath.Create(0, args.OldStartingIndex);
+				var newPath = NSIndexPath.Create(0, args.NewStartingIndex);
+
+				_collectionView.MoveItem(oldPath, newPath);
+				return;
+			}
+
+			var start = Math.Min(args.OldStartingIndex, args.NewStartingIndex);
+			var end = Math.Max(args.OldStartingIndex, args.NewStartingIndex) + count;
+			_collectionView.ReloadItems(CreateIndexesFrom(start, end));
 		}
 		
 		private void Replace(NotifyCollectionChangedEventArgs args)
 		{
 			var startIndex = args.NewStartingIndex > -1 ? args.NewStartingIndex : _itemsSource.IndexOf(args.NewItems[0]);
```

Comment:
this var need only for `ReloadItems`. Please move it into `if` block below.

Reply:
Fixed.